ecosystem/sep-0040.md

```diff
@@ -61,12 +61,12 @@ pub trait PriceFeedTrait {
     /// Return the base asset the price is reported in
     fn base(
         env: soroban_sdk::Env,
-    ) -> Address;
+    ) -> Bytes;
```

Comment:
I would recommend using descriptive UDTs instead of Bytes and custom encodings.

For example, if your goal is to represent on-chain assets as well as off-chain entities, I would recommend to use the following enum to represent assets:

```rust
[contracttype]
enum Asset {
   Contract(Address),
   Generic(Bytes/String)
}
```

This will provide much greater readability and reduce the risk of introducing bugs.

Reply:
I agree

Reply:
Does Soroban support `String` type? If yes, then I think we can use `String`, otherwise `Bytes` seems to be best option to avoid storing unnecessary information.

Reply:
> Does Soroban support String type?

Yes: https://docs.rs/soroban-sdk/latest/soroban_sdk/struct.String.html

Reply:
> We also have to consider the performance/compute cost of using a enum instead of raw Bytes. 

It's actually likely cheaper to use UDT. Don't forget, that you need to do custom decoding of your bytes input, which is more or less what you'd end up doing with enums (but for the Soroban UDTs you're guaranteed to be using the host-based decoding, which should be fast).